tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -93,13 +135,154 @@ public async Task<long> UploadRepositoryChangesAsync()
+            {
+                if (finalTry)
+                {
+                    Log.Error<int, string>("Failed to get settings with status code {StatusCode} and message: {ResponseContent}", response.StatusCode, responseContent);
+                }
+
+                throw new WebException($"Status: {response.StatusCode}, Content: {responseContent}");
+            }
+
+            Log.Debug("ITR: JSON RS = {json}", responseContent);
+            var deserializedResult = JsonConvert.DeserializeObject<DataEnvelope<Data<SettingsResponse>?>>(responseContent);
+            return deserializedResult.Data?.Attributes ?? default;
+        }
+    }
+
+    public async Task<SkippableTest[]> GetSkippableTestsAsync()
+    {
+        Log.Debug("ITR: Getting skippable tests...");
+        var framework = FrameworkDescription.Instance;
+        var repository = await _getRepositoryUrlTask.ConfigureAwait(false);
+        var currentSha = await _getShaTask.ConfigureAwait(false);
+        if (currentSha is null)
+        {
+            Log.Warning("ITR: 'git rev-parse HEAD' command is null");
+            return Array.Empty<SkippableTest>();
+        }
+
+        currentSha = currentSha.Replace("\n", string.Empty);
+
+        var query = new DataEnvelope<Data<SkippableTestsQuery>>(
+            new Data<SkippableTestsQuery>(
+                default,
+                TestParamsType,
+                new SkippableTestsQuery(
+                    _serviceName,
+                    _environment,
+                    repository,
+                    currentSha,
+                    new TestsConfigurations(
+                        framework.OSPlatform,
+                        Environment.OSVersion.VersionString,
+                        framework.OSArchitecture,
+                        framework.Name,
+                        framework.ProductVersion,
+                        framework.ProcessArchitecture))),
+            default);
+        var jsonQuery = JsonConvert.SerializeObject(query, SerializerSettings);
+        var jsonQueryBytes = Encoding.UTF8.GetBytes(jsonQuery);
+        Log.Debug("ITR: JSON RQ = {json}", jsonQuery);
+
+        return await WithRetries(InternalGetSkippableTestsAsync, jsonQueryBytes, MaxRetries).ConfigureAwait(false);
+
+        async Task<SkippableTest[]> InternalGetSkippableTestsAsync(byte[] state, bool finalTry)
+        {
+            var request = _apiRequestFactory.Create(_skippableTestsUrl);
+            request.AddHeader(ApiKeyHeader, _settings.ApiKey);
+            request.AddHeader(ApplicationKeyHeader, _settings.ApplicationKey);
+            request.AddHeader(HttpHeaderNames.TraceId, _id);
+            request.AddHeader(HttpHeaderNames.ParentId, _id);
+            Log.Debug("ITR: Searching skippable tests from: {url}", _skippableTestsUrl.ToString());
+            var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
```

Comment:
```suggestion
            using var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
```

Reply:
done in [6bc5a57](https://github.com/DataDog/dd-trace-dotnet/pull/3075/commits/6bc5a574614dc13585e91dedf1c8bd721bd4603a)